### PR TITLE
#162163281 users should be able to bookmark articles for reading later

### PR DESCRIPTION
**Pull request by Kasulejoseph**
**What does this PR do?**
The PR delivers a feature for users to bookmark reading articles 

**Description of the tasks to be completed?**
- Have these endpoints working
- POST/GET `/api/article/:slug/bookmark/` --> For bookmarking and getting user bookmarks
- DELETE `/api/article/bookmark/:id/`  --> To unbookmark an article

**How should this be tested?**
```- Clone repository
- create a virtual environment
- Install requirements `pip install -r requirements.txt`
- Run the project `python manage.py runserver 
```
**To bookmark an article**
Run this endpoint in postman with a POST method `/api/article/:slug/bookmark/`
_ie_ use a valid article slug
`output`
screen shot 2018-12-20 at 02 04 02

**To undo bookmarking**
In postman, navigate to DELETE `/api/article/bookmark/id/`
_ie_ `id` is the id of the bookmark
`output`
screen shot 2018-12-20 at 02 06 03

**To List user bookmarks**
To view all the users recent bookmarks, navigate to `/api/article/:slug/bookmark/` 
with A GET method. this should return the user bookmarks.
screen shot 2018-12-20 at 15 40 51

- user can only bookmarks an article once
screen shot 2018-12-20 at 02 04 50

**Relavant story**
- 162163281

### CODE REVIEW

**Review by bibangamba — APPROVED** [review]
nice work

**Review by Nta1e — APPROVED** [review]

**Comment by Nta1e** [comment]
👏🏽

**Review by joshtrigger — APPROVED** [review]

File: authors/apps/articles/test/test_bookmark.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            '/api/articles/', data=self.article, format='json')
+        slug = 'Fake slug'
+        res = self.client.post(
+            '/api/articles/{}/bookmark/'.format(slug),
+            format='json'
+        )
+        self.assertEqual(res.status_code, 404)
+        self.assertEqual("Article does not exist", res.data['error'])
+
+    def test_get_bookmarked_articles(self):
```
Comment (collin5) [inline authors/apps/articles/test/test_bookmark.py]: Looks like this tests getting a single bookmark. Should be `test_get_bookmarked_article`

**Review by collin5 — CHANGES_REQUESTED** [review]
Looks good! Can you please also add an endpoint for getting all bookmarks `/api/article/bookmarks/`. Also don't forget to test this.

File: authors/apps/articles/test/test_bookmark.py (hunk 1 above)
Comment (Kasulejoseph) [reply authors/apps/articles/test/test_bookmark.py]: thanks. I have added the right endpoint for getting all the user bookmarks

**Review by collin5 — APPROVED** [review]

**Pull request merged**